index.html

```diff
@@ -0,0 +1,571 @@
+
+Standard extensibility rules:
+- **Directives** may be defined as taking additional arguments, provided those arguments are [prefixed](#prefixing) with the name of a document feature followed by two underscores ("`__`"). These arguments are interpreted as metadata-on-metadata, and are routed to the declaring spec for interpretation
+- **Enums** defined in features may include values of any name. For extensible enums, the space of **unprefixed** names belongs to the document. The space of **prefixed** names belongs to the spec with that prefix, and may be used to pass data between specs.
+- **Input Objects** may be defined with additional fields, provided those fields are [prefixed](#prefixing) with the name of a feature followed by two underscores.
+
+## Processing Schemas
+
+```mermaid
+#::[Example processing pipeline]
+graph LR
+  schema(["📄  Input Schema"]):::file-->proc("🤖 &nbsp;Processor")
+  proc-->output(["📄  Output Schema"]):::file
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common use case is that of a processor which consumes a valid input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - **should** pass through `@core` directives which reference unknown feature URLs
+  - **should** pass through prefixed directives, types, and other schema elements  
+
+An exception to this is processors which prepare the schema for final public consumption. Such processors **may** choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+## Validations &amp; Algorithms
+
+### Bootstrapping *validates* Has Core Feature
+
+Determine the name of the core specification within the document *or* fail the *Has Core Feature* validation.
+
+It is possible to [rename the core feature](#renaming-core-itself) within a document. This process determines the actual name for the core feature if one is present.
+
+**Formal Specification**
+- For each directive `d` on the SchemaDefinition within the Document,
+  - If `d` has a `feature:` argument whose value is "https://lib.apollo.dev/core/v0.1", *and either*:
+    - &#8230;`d` has an `as:` argument whose value is equal to `d`'s name
+    - &#8230;*or* `d` does not have an `as:` argument and `d`'s name is `core`
+    - *then* **Return** `coreName` = `d`'s name
+- If no matching directive was found, the ***Has Core Feature* validation fails**.
+
+### Namespace Collection *validates* Name Uniqueness
```

Comment:
This is the first reference to "namespaces" as a thing (vs just the concept of a "namespace separator". I think this maybe should be returning `featureDirectives` rather than `namespaces`? Or `featureDirectivesByName`?

Reply:
Changed to `CollectFeatures` and `AssignFeatures`.